File: result/metagpt/claude/humaneval/110.c
#include <stdio.h>
#include <string.h>

const char* exchange(int* lst1, int lst1_size, int* lst2, int lst2_size) {
    int num = 0;
    for (int i = 0; i < lst1_size; i++) {
        if (lst1[i] % 2 == 0) num++;
    }
    for (int i = 0; i < lst2_size; i++) {
        if (lst2[i] % 2 == 0) num++;
    }
    if (num >= lst1_size) return "YES";
    return "NO";
}

File: result/metagpt/gpt4o/humaneval/110.rs
fn exchange(lst1: &[i32], lst2: &[i32]) -> &'static str {
    let mut num = 0;
    for &item in lst1.iter() {
        if item % 2 == 0 {
            num += 1;
        }
    }
    for &item in lst2.iter() {
        if item % 2 == 0 {
            num += 1;
        }
    }
    if num >= lst1.len() {
        "YES"
    } else {
        "NO"
    }
}

fn main() {
    let lst1 = vec![1, 2, 3, 4];
    let lst2 = vec![5, 6, 7, 8];
    let result = exchange(&lst1, &lst2);
    println!("Exchange result: {}", result);
}